tests/utilities_tests.cpp

```diff
@@ -0,0 +1,84 @@
+//==============================================================================
+// Copyright (c) 2019-2022 CradleApps, LLC - All Rights Reserved
+//
+// This file is part of the Cradle Engine. Unauthorised copying and
+// redistribution is strictly prohibited. Proprietary and confidential.
+//==============================================================================
+
+#include "realtime_memory/utilities.h"
+
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+#include <catch2/generators/catch_generators_random.hpp>
+#include <catch2/generators/catch_generators_range.hpp>
+
+//==============================================================================
+TEST_CASE ("merge sort linked lists")
+{
+    struct node
+    {
+        node* next = nullptr;
+        int data = 0;
+
+        explicit node (int d) : data (d) {}
+    };
+
+    const int listLength = 1000;
```

Comment:
`listLength` is slightly misleading here as there end up being `listLength + 1` nodes in the list making the test section `CHECK`s slightly surprising to read in isolation (you have to look further up the test to understand that the `listLength` is not actually the list length. Not sure it's critical to change, but thought I'd point it out.